editor/src/components/canvas/remix/remix-utils.tsx

```diff
@@ -10,3 +21,66 @@ export const DefaultFutureConfig: FutureConfig = {
   v2_normalizeFormMethod: false,
   v2_routeConvention: true,
 }
+
+export interface EntryRouteWithFilePath extends EntryRoute {
+  filePath: string
+}
+
+// This is necessary to create a simple node.fs-like implementation for Utopia projectContents, which
+// can be used by the Remix functions to parse the routes
+export function projectContentsToFileOps(projectContents: ProjectContentTreeRoot): FileOps {
+  return {
+    existsSync: (file: string): boolean => getContentsTreeFromPath(projectContents, file) != null,
+    readdirSync: (dir: string): Array<string> => {
+      const projectDir = getContentsTreeFromPath(projectContents, dir)
+      let entries =
+        projectDir != null && projectDir.type === 'PROJECT_CONTENT_DIRECTORY'
+          ? Object.values(projectDir.children).map((tree) => `/${tree.fullPath}`)
+          : []
+      return entries
+    },
+    isDirectory: (file: string): boolean => {
+      const projectFile = getContentsTreeFromPath(projectContents, file)
+      return projectFile != null && projectFile.type === 'PROJECT_CONTENT_DIRECTORY'
+    },
+    isFile: (file: string): boolean => {
+      const projectFile = getContentsTreeFromPath(projectContents, file)
+      return projectFile != null && projectFile.type === 'PROJECT_CONTENT_FILE'
+    },
+  }
+}
+
+export function createRouteManifestFromProjectContents(
+  projectContents: ProjectContentTreeRoot,
+): RouteManifest<EntryRouteWithFilePath> | null {
+  const routesFromRemix = flatRoutes(ROOT_DIR, projectContentsToFileOps(projectContents))
+
+  if (routesFromRemix == null) {
+    return null
+  }
+  return patchRemixRoutes(routesFromRemix)
+}
+
+function patchRemixRoutes(routesFromRemix: RouteManifest<ConfigRoute> | null) {
+  const routesFromRemixWithRoot: RouteManifest<ConfigRoute> = {
+    ...routesFromRemix,
+    root: { path: '', id: 'root', file: 'root.js', parentId: '' },
```

Comment:
Is this hardcoding expected / required, or are we going to remove this at some point?